src/transformStateWithClones.js

```diff
@@ -70,7 +70,29 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, transforms) {
-  // write code here
+  const newState = { ...state };
+  const result = [];
+
+  transforms.forEach(function(item, i, arr) {
+    if (item.operation === 'addProperties') {
+      for (const key in item.properties) {
+        newState[key] = item.properties[key];
+      };
+    };
+    if (item.operation === 'removeProperties') {
+      for (const value of item.properties) {
+        delete newState[value];
+      };
+    };
+    if (item.operation === 'clear') {
+      for (const key in newState) {
+        delete newState[key];
+      };
+    };
+    const obj = { ...newState };
+    result.push(obj);
```

Comment:
why not use the spread operator at once?
```suggestion
    result.push(...newState);
```